core/src/main/java/org/apache/carbondata/core/view/MVProvider.java

```diff
@@ -121,70 +109,69 @@ private SchemaProvider getSchemaProvider(String databaseName) {
+  public MVSchema getSchema(MVManager viewManager, String databaseName, String viewName)
+      throws IOException {
+    SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);
     if (schemaProvider == null) {
       return null;
     }
     return schemaProvider.retrieveSchema(viewManager, viewName);
   }
 
-  List<MVSchema> getSchemas(MVManager viewManager,
-                            String databaseName, CarbonTable carbonTable) throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  List<MVSchema> getSchemas(MVManager viewManager, String databaseName, CarbonTable carbonTable)
+      throws IOException {
+    SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);
     if (schemaProvider == null) {
       return Collections.emptyList();
     } else {
       return schemaProvider.retrieveSchemas(viewManager, carbonTable);
     }
   }
 
-  List<MVSchema> getSchemas(MVManager viewManager,
-                            String databaseName) throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  List<MVSchema> getSchemas(MVManager viewManager, String databaseName) throws IOException {
+    SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);
     if (schemaProvider == null) {
       return Collections.emptyList();
     } else {
       return schemaProvider.retrieveAllSchemas(viewManager);
     }
   }
 
-  void saveSchema(MVManager viewManager, String databaseName,
-                  MVSchema viewSchema) throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  void saveSchema(MVManager viewManager, String databaseName, MVSchema viewSchema)
+      throws IOException {
+    SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);
     if (schemaProvider == null) {
       throw new IOException("Database [" + databaseName + "] is not found.");
     }
     schemaProvider.saveSchema(viewManager, viewSchema);
   }
 
-  public void dropSchema(String databaseName, String viewName)
+  public void dropSchema(MVManager viewManager, String databaseName, String viewName)
       throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+    SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);
     if (schemaProvider == null) {
       throw new IOException("Materialized view with name " + databaseName + "." + viewName +
           " does not exists in storage");
     }
     schemaProvider.dropSchema(viewName);
   }
 
-  private String getStatusFileName(String databaseName) {
+  private String getStatusFileName(MVManager viewManager, String databaseName) {
     if (databaseName.equalsIgnoreCase("default")) {
-      return this.storeLocation +
+      return viewManager.getDatabaseLocation(databaseName) +
```

Comment:
Move viewManager.getDatabaseLocation(databaseName) before ifelse and reuse

Reply:
OK